src/puppetlabs/trapperkeeper/services/webserver/jetty9_core.clj

```diff
@@ -952,6 +956,25 @@
                                             [server-id (initialize-context)]))
                  :default-server (get-default-server-from-config config)))
 
+(schema/defn ^:always-validate reload-crl-on-change!
+  "Reload the CRL file used by the supplied ssl-context-factory whenever any
+  changes to the file occur."
+  [ssl-context-factory :- InternalSslContextFactory
+   watcher :- (schema/protocol watch-protocol/Watcher)]
+  (when-let [crl-path (.getCrlPath ssl-context-factory)]
+    (let [normalized-crl-path (.getCanonicalPath (fs/file crl-path))]
+      (watch-protocol/add-watch-dir! watcher (fs/parent crl-path)
```

Comment:
Having to do a recursive watch of all of the files in the directory that the CRL file resides in is definitely not ideal but this is all which is available in the current trapperkeeper-filesystem-watcher service.  At the very least, it should be possible to enhance the watcher service to only watch the immediate directory rather than recursively watching subdirectories.  Even better would be to be able to just watch the CRL file itself but it's not clear to me that the underlying [Path.register](https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html#register-java.nio.file.WatchService-java.nio.file.WatchEvent.Kind...-) method which is used by the service would support watching an individual file - as opposed to a directory.